sql/item_float_interchange.h

```diff
@@ -0,0 +1,130 @@
+  }
+
+ public:
+  Item_func_interchange_float32_to_uint32(THD *thd, Item *a)
+      : Item_long_func(thd, a) {}
+  ulonglong val_uint();
+  longlong val_int();
+  const char *func_name() const { return "interchange_float32_to_uint32"; }
+  bool fix_length_and_dec() {
+    decimals= 0;
+    max_length= MAX_INT_WIDTH;
+    maybe_null= args[0]->maybe_null;
+    unsigned_flag= 1;
+    return FALSE;
+  }
+  Item *get_copy(THD *thd) {
+    return get_item_copy<Item_func_interchange_float32_to_uint32>(thd, this);
+  }
+};
+
+/*************************************************************************
+  Item_func_interchange_uint32_to_float32
+*************************************************************************/
+
+class Item_func_interchange_uint32_to_float32 : public Item_real_func {
+  bool check_arguments() const {
+    return check_argument_types_can_return_int(0, arg_count);
+  }
+
+ public:
+  Item_func_interchange_uint32_to_float32(THD *thd, Item *a)
+      : Item_real_func(thd, a) {}
+  double val_real();
+  const char *func_name() const { return "interchange_uint32_to_float32"; }
+  bool fix_length_and_dec() {
+    maybe_null= args[0]->maybe_null;
+    decimals= NOT_FIXED_DEC;
+    return FALSE;
+  }
+  Item *get_copy(THD *thd) {
+    return get_item_copy<Item_func_interchange_uint32_to_float32>(thd, this);
+  }
+};
+#endif  // float32_exists
+
+
+#if float64_exists
+/*************************************************************************
+  Item_func_interchange_float64_to_uint64
+*************************************************************************/
+
+class Item_func_interchange_float64_to_uint64 : public Item_longlong_func {
+  bool check_arguments() const {
+    return check_argument_types_can_return_real(0, arg_count);
+  }
+
+ public:
+  Item_func_interchange_float64_to_uint64(THD *thd, Item *a)
+      : Item_longlong_func(thd, a) {}
+  ulonglong val_uint();
+  longlong val_int();
+  const char *func_name() const { return "interchange_float64_to_uint64"; }
+  bool fix_length_and_dec() {
+    decimals= 0;
+    max_length= MAX_BIGINT_WIDTH;
+    maybe_null= args[0]->maybe_null;
+    unsigned_flag= 1;
+    return FALSE;
+  }
+  Item *get_copy(THD *thd) {
+    return get_item_copy<Item_func_interchange_float64_to_uint64>(thd, this);
+  }
+};
+
+/*************************************************************************
+  Item_func_interchange_uint64_to_float64
+*************************************************************************/
+
+class Item_func_interchange_uint64_to_float64 : public Item_real_func {
+  bool check_arguments() const {
+    return check_argument_types_can_return_int(0, arg_count);
+  }
+
+ public:
+  Item_func_interchange_uint64_to_float64(THD *thd, Item *a)
+      : Item_real_func(thd, a) {}
+  double val_real();
+  const char *func_name() const { return "interchange_uint64_to_float64"; }
+  bool fix_length_and_dec() {
+    maybe_null= args[0]->maybe_null;
```

Comment:
maybe_null= true;

(sorry again)

Reply:
@abarkov okay, will do. also wlad had had some very valuable input on IRC #maria which convinced me I should (1) go signed, rather than unsigned, because that is nicer for connectors and (2) that no platforms that mariadb-server targets have "double" and "float" defined to be other than 64 and 32 bits, thus we do not need an abstraction of types to "float64" and "float32" nor the pre-processor if-exists kinds of directives.

I am re-working the patch to take this suggestions on-board.